File: 1971.py
class Solution:
    def validPath(self, n: int, edges: List[List[int]], source: int, destination: int) -> bool:
        graph={}
        for i in edges:
            if i[0] in graph:
                graph[i[0]].append(i[1])
            else: graph[i[0]]=[i[1]]
            
            if i[1] in graph:
                graph[i[1]].append(i[0])
            else: graph[i[1]]=[i[0]]
                
        isvisited=[False]*n
        q=[]
        q.append(source)
        while q:
            nodeval=q.pop(0)
            isvisited[nodeval]=True
            if isvisited[destination]: return True
            for i in graph[nodeval]:
                if not isvisited[i]: q.append(i)
            
        return False
        

File: 1971.cpp
class Solution {
public:
    bool validPath(int n, vector<vector<int>>& edges, int source, int destination) {
        unordered_map<int,vector<int>> graph;
        for(auto e:edges)
        {
            graph[e[0]].push_back(e[1]);
            graph[e[1]].push_back(e[0]);
        }
        vector<bool> isvisited(n,false);
        queue<int> q;
        q.push(source);
        while(!q.empty())
        {
            int nodeval=q.front();
            isvisited[nodeval]=true;
            q.pop();
            if(isvisited[destination]) return true;
            for(auto connect:graph[nodeval])
            {
                if(!isvisited[connect]) q.push(connect);
            }
        }
        return false;
    }
};
